### PR TITLE
use Enum start number

### PR DESCRIPTION
**Pull request by gyorb**
Just a small improvement to use the Enum start number available since python 3.5

part of #3020

---
This change is [Reviewable](https://reviewable.io/reviews/qutebrowser/qutebrowser/3191)

### CODE REVIEW

**Comment by gyorb** [comment]
I tried to search for more places where the Enum start could be used or adds some improvement but most of the enums are build in the userttypes module and not used directly.
Let me know if I you know about more places it could be improved.

The general rule is to generate an enum through the userttypes modul?

**Comment by The-Compiler** [comment]
I was actually thinking about that... The main reason `usertypes.Enum` exists is from [ancient times](https://github.com/qutebrowser/qutebrowser/commit/4ebe643ea668a9c6cfa0b512424cd067c0f922d3) when qutebrowser still supported Python 3.3 which didn't have `enum.Enum`. 

Somewhat later, 3.3 support [was dropped](https://github.com/qutebrowser/qutebrowser/commit/8d80ce26282f42989719541448475854c0266d67), but `usertypes.enum` still persisted to support things like `start`.

Nowadays I think we really should just use `enum.Enum` instead (probably the [functional API](https://docs.python.org/3/library/enum.html#functional-api) so we don't have to specify values). Is this something you'd like to work on?

**Comment by gyorb** [comment]
@The-Compiler sure, I can update the code to use `enum.Enum` or `enum.IntEnum` instead of the current `usertypes.enum`

**Comment by gyorb** [comment]
I removed the `usertypes.enum` from the source. The uniqueing check for the enums is missing. It could be done like this for each enum initialization:
```
ModelRole = enum.IntEnum('ModelRole', ['item'], start=Qt.UserRole)
ModelRole = enum.isunique(ModelRole)
```
It adds some extra code which I do not think is always necessary, there are many simple enums. If it is needed I can add the uniqueing check. What do you think?

File: qutebrowser/browser/hints.py [hunk 1]
```diff
@@ -37,10 +38,10 @@
...
 
 
-Target = usertypes.enum('Target', ['normal', 'current', 'tab', 'tab_fg',
-                                   'tab_bg', 'window', 'yank', 'yank_primary',
-                                   'run', 'fill', 'hover', 'download',
-                                   'userscript', 'spawn'])
+Target = enum.Enum('Target', ['normal', 'current', 'tab', 'tab_fg',
+                              'tab_bg', 'window', 'yank', 'yank_primary',
+                              'run', 'fill', 'hover', 'download',
+                              'userscript', 'spawn'])
```
Comment (The-Compiler) [inline qutebrowser/browser/hints.py]: You can re-flow this now that the line is shorter.

File: qutebrowser/mainwindow/statusbar/url.py [hunk 2]
```diff
@@ -27,8 +29,8 @@
 
 
 # Note this has entries for success/error/warn from widgets.webview:LoadStatus
-UrlType = usertypes.enum('UrlType', ['success', 'success_https', 'error',
-                                     'warn', 'hover', 'normal'])
+UrlType = enum.Enum('UrlType', ['success', 'success_https', 'error',
+                                'warn', 'hover', 'normal'])
```
Comment (The-Compiler) [inline qutebrowser/mainwindow/statusbar/url.py]: You can re-flow this now that the line is shorter.

File: qutebrowser/mainwindow/tabwidget.py [hunk 3]
```diff
@@ -34,8 +35,8 @@
 from qutebrowser.misc import objects
 
 
-PixelMetrics = usertypes.enum('PixelMetrics', ['icon_padding'],
-                              start=QStyle.PM_CustomBase, is_int=True)
+PixelMetrics = enum.IntEnum('PixelMetrics', ['icon_padding'],
+                         start=QStyle.PM_CustomBase)
```
Comment (The-Compiler) [inline qutebrowser/mainwindow/tabwidget.py]: this should be indented to be under the `'` again.

File: qutebrowser/utils/usertypes.py [hunk 4]
```diff
@@ -25,7 +25,7 @@
 
 import operator
 import collections.abc
-import enum as pyenum
+from enum import Enum, IntEnum
```
Comment (The-Compiler) [inline qutebrowser/utils/usertypes.py]: Please do `import enum` here, and use `enum.Enum` etc. here - I don't like importing types from modules, as it makes it difficult to keep track what's coming from where.

**Comment by gyorb** [comment]
I've updated it, should I do any other changes?

**Comment by The-Compiler** [comment]
Thank you!

**Pull request merged**